VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -99,19 +99,19 @@ IOReturn VoodooI2CHIDDevice::parseHIDDescriptor() {
 
 IOReturn VoodooI2CHIDDevice::getHIDDescriptorAddress() {
     uuid_t guid;
-    uuid_parse(HID_I2C_DSM_HIDG, guid);
+    uuid_parse(I2C_DSM_HIDG, guid);
 
     // convert to mixed-endian
-    *(uint32_t *)guid = OSSwapInt32(*(uint32_t *)guid);
-    *((uint16_t *)guid + 2) = OSSwapInt16(*((uint16_t *)guid + 2));
-    *((uint16_t *)guid + 3) = OSSwapInt16(*((uint16_t *)guid + 3));
+    *(reinterpret_cast<uint32_t *>(guid)) = OSSwapInt32(*(reinterpret_cast<uint32_t *>(guid)));
```

Comment:
Please avoid using `reinterpret_cast` or any of the other C++ casts. 

The mach kernel has it's own type definition table and it's highly recommended to use `OSDynamicCast` instead.

See documentation here:
https://developer.apple.com/documentation/driverkit/osdynamiccast

Reply:
I changed that since VoodooI2C main project prompts so at linting stage. However, in libkern/libkern/c++/OSObject.h, it says "OSDynamicCast is build on the OSMetaClass mechanism". Or if the driverkit's version got some enhancement?

Reply:
You are correct. This has nothing to do with OS class definitions as OSSwapInt32 converts kernel specific numbers into integers. Please ignore the previous comment.